enhancements/ebpf-in-ocp.md

```diff
@@ -0,0 +1,211 @@
+---
+title: ebpf-in-ocp
+authors:
+  - "@danwinship"
+reviewers:
+  - "@TBD, for network observability"
+  - "@TBD, for other existing eBPF use case X"
+  - "@TBD, for other future eBPF use case Y"
+  - "@dave-tucker, for bpfd"
+approvers:
+  - TBD
+api-approvers:
+  - None
+creation-date: 2022-05-25
+last-updated: 2022-05-25
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - TBD
+see-also:
+replaces:
+superseded-by:
+---
+
+# Guidelines for the Use of eBPF in OCP
+
+## Summary
+
+OCP developers and customers alike would like to use eBPF in OCP
+clusters. At the present time, _some_ eBPF hooks are designed to be
+"shareable", but others are not (and some are shareable only when used
+in certain restricted ways). We need to create rules for how OCP will
+use eBPF and how we expect others to use (or not use it) to prevent
+conflicts later.
+
+## Motivation
+
+### User Stories
+
+#### Existing Use Case Number 1
+
+(Something about observability or tracing where people are already
+using eBPF in OCP.)
+
+#### Upcoming Network Observability Features
+
+(Something about upcoming network observability features.)
```

Comment:
```suggestion
As an OCP developer, I want to attach the eBPF agent into different parts of
the system (e.g. Nodes' kernel) to be able to trace the network flows within
the system, as well as being able to correlate them with other parts of the
system configuration (e.g. Network Policies, security, etc.).
```

Reply:
should we also describe the end user (OCP customer) use cases that we address? Something like: "as an OCP customer, I want to be able to trace network flows across pods and services in my cluster, with a resource overhead as low as possible." ?

Reply:
I don't think we need to get into the higher-level goals that led to people using eBPF... for purposes of this enhancement, we're starting with "I want to use eBPF to do X" and not really worrying about how you got to that point. At least for OCP-internal use cases. For customer use cases we may need to distinguish between "customer wants to use eBPF and we think it's reasonable" vs "customer wants to use eBPF and we think it's a bad idea". (eg, that's why I separated out the "eBPF for bugfixes" and "eBPF for new features" use cases, because we might have different opinions on them)